Lib/tempfile.py

```diff
@@ -421,10 +421,11 @@ class _TemporaryFileCloser:
     file = None  # Set here since __del__ checks it
     close_called = False
 
-    def __init__(self, file, name, delete=True):
+    def __init__(self, file, name, delete=True, delete_on_close=True):
```

Comment:
Nothing needs to change in this class. Instead, change the way it gets instantiated in `_TemporaryFileWrapper.__init__()` as follows: `_TemporaryFileCloser(file, name, delete and delete_on_close)`. The meaning of `delete` in the closer has always been "delete on close", and that won't change.

Reply:
This is now implemented in[ this commit](https://github.com/python/cpython/pull/97015/commits/6944ab23e513861508fe643558efce47f178875e)